Utils/azuretools-core/src/com/microsoft/azuretools/entity/BaseEntity.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright (c) Microsoft Corporation
+ *
+ * All rights reserved.
+ *
+ * MIT License
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated
+ * documentation files (the "Software"), to deal in the Software without restriction, including without limitation
+ * the rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, and
+ * to permit persons to whom the Software is furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all copies or substantial portions of
+ * the Software.
+ *
+ * THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO
+ * THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
+ * TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package com.microsoft.azuretools.entity;
+
+import org.apache.commons.lang3.builder.EqualsBuilder;
+import org.apache.commons.lang3.builder.HashCodeBuilder;
+import org.apache.commons.lang3.builder.ReflectionToStringBuilder;
+import org.apache.commons.lang3.builder.ToStringStyle;
+
+import java.io.Serializable;
+
+/**
+ * @author qianjinshen
+ * @date 2020-09-24 14:45:34
+ */
+public abstract class BaseEntity implements Serializable {
```

Comment:
I didn't get what the class is used for, the name is too general.

Reply:
general/abstract is the key of this class. it should to let any sub class have default implements of hashCode&equals&toString.

Reply:
I'm still not getting the point of why we need this base class. Looks like it's a Kotlin data class implementation https://kotlinlang.org/docs/reference/data-classes.html ?

Reply:
Yes. Ketlin does have same feature in a degree.
Example for toString:
if a classs such as User extends the BaseEntity, the return value of its toString method will be like "User(name=xxx, age=xxx)....".
or, its value should be just like "package.User@7de26db8".